src/ros/controller.cpp

```diff
@@ -33,7 +33,10 @@ void ROSController::doSwitch(const std::list<hardware_interface::ControllerInfo>
 
   for (auto const& ci : start_list)
   {
-    auto ait = available_interfaces_.find(ci.name);
+    if (ci.claimed_resources.empty())
+      continue;
+
+    auto ait = available_interfaces_.find(ci.claimed_resources[0].hardware_interface);
```

Comment:
This doesn't work on Indigo which was a requirement when I wrote the code, I don't know if it's reasonable to drop Indigo now?
It's not really my decision to make either as ROS Industrial is supposed to adopt the code eventually if my branch is every merged into the mainline.

Reply:
In my understanding this branch was meant to replace ur_modern_driver (and hopefully ur_driver) in kinetic+, so I did not consider indigo compatibility.

Anyway, ros_control only generalized their code from single hardware interface to lists of hardware interfaces.
I just added a preprocessor workaround to handle both cases. We use the same workaround [in moveit](https://github.com/ros-planning/moveit/blob/kinetic-devel/moveit_plugins/moveit_ros_control_interface/CMakeLists.txt#L24).